src/vsphere_cpi/spec/unit/cloud/vsphere/drs_rules/drs_rule_spec.rb

```diff
@@ -151,4 +151,48 @@ def self.it_adds_custom_attribute_to_vm
     end
   end
 
+  describe '#add_vm_host_affinity_rule' do
+    let(:vm) { instance_double('VimSdk::Vim::VirtualMachine', name: 'fake-vm') }
+    let(:fake_vm_1) { instance_double('VimSdk::Vim::VirtualMachine', name: 'fake-vm-1') }
+    let(:vm_group_name) {'fake-vm-group'}
+    let(:host_group_name) {'fake-host-group'}
+    let(:configuration_ex) { double(:configuration_ex, rule: rules, group: vm_groups) }
+    let(:vm_groups) { [ VimSdk::Vim::Cluster::VmGroup.new(name: vm_group_name, vm: [fake_vm_1]) ] }
+    let(:rules) { [ double(:rule, name: 'fake-rule-name', key: rule_key) ] }
+    let(:rule_key) { 33 }
+
+    before do
+      allow(datacenter_cluster).to receive(:configuration_ex).and_return(configuration_ex)
+      allow(datacenter_cluster).to receive(:reconfigure_ex).and_return(task)
+    end
+
+    context 'when rule exists' do
+      it 'returns the rule with DRS lock' do
+        expect(drs_lock).to receive(:with_drs_lock).and_yield.ordered
+        expect(datacenter_cluster).to_not receive(:reconfigure_ex)
+        expect(drs_rule).to_not receive(:create_vm_host_affinity_rule)
+        drs_rule.add_vm_host_affinity_rule(vm, vm_group_name, host_group_name)
+      end
+    end
+    context 'when rule does not exist' do
+      let(:rules) { [] }
+      it 'adds vm to the vm_group and creates DRS rule with DRS lock' do
```

Comment:
reword, as this method only creates vm_host_afffinity_rule and does not add vm to vm_group